src/partitioning/partitioner.C

```diff
@@ -976,7 +976,13 @@ void Partitioner::build_graph (const MeshBase & mesh)
   std::vector<const Elem *> neighbors_offspring;
 #endif
 
-   if (!_global_index_by_pid_map.size() || _global_index_by_pid_map.size() != mesh.n_active_elem())
+   bool need_global_index_by_pid_map =
+     !_global_index_by_pid_map.size() ||
+     _global_index_by_pid_map.size() != mesh.n_active_elem();
+
+   mesh.comm().max(need_global_index_by_pid_map);
+
+   if (need_global_index_by_pid_map)
```

Comment:
Talked to @jwpeterson  a little on this. It looks like for me the best solution is remove `if` totally. Then we always construct the global map.

Reply:
There may be some cases we can not reuse map even the n_ative_elems does not change.  For example, one core does a refinement while another core does coarsening.